Sources/Core/AWSClientRuntime/Endpoints/EndpointResolverMiddleware.swift

```diff
@@ -0,0 +1,131 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import class Smithy.Context
+import struct SmithyHTTPAPI.Endpoint
+import class SmithyHTTPAPI.SdkHttpRequest
+import class SmithyHTTPAPI.SdkHttpRequestBuilder
+import struct SmithyHTTPAuthAPI.SelectedAuthScheme
+import enum SmithyHTTPAuthAPI.SigningAlgorithm
+import enum SmithyHTTPAuthAPI.SigningPropertyKeys
+import protocol ClientRuntime.ApplyEndpoint
+import struct ClientRuntime.DefaultEndpointsAuthSchemeResolver
+import enum ClientRuntime.EndpointsAuthScheme
+import protocol ClientRuntime.EndpointsAuthSchemeResolver
+import protocol ClientRuntime.EndpointsRequestContextProviding
+import protocol ClientRuntime.Handler
+import protocol ClientRuntime.Middleware
+import struct ClientRuntime.OperationOutput
+
+public struct EndpointResolverMiddleware<OperationStackOutput, Params: EndpointsRequestContextProviding>: Middleware {
```

Comment:
This is the AWS-specific reusable endpoint middleware.  See the same type in smithy-swift for details.

This differs from the smithy-swift version in that it also handles AWS region.